### PR TITLE
Refactor xetex_aatfont's dependencies

### LINKED ISSUE #36: bring back XETEX_MACOS code sections
Mainly some Mac-specific font handling.

### PR DESCRIPTION
**Pull request by cormacrelf**
- [x] use the core_foundation crates as much as possible (they aren't complete)
- [x] `#ifdef XETEX_MAC` as a macro
- [x] replace the c2rust FreeType externs with freetype-rs
- [x] have gFreeTypeLibrary shared with the cpp? Nah. currently in thread local, not interacting with the static var. This is probably ok, just something to think about when porting the cpp (use the same thread local).
- [ ] clean up `xetex_ext.rs` again as I kinda ruined some of it
- [ ] test it on linux or windows (I can't) (may be missing a couple of items/structs from the top?)

It's a little noisy, partly because the original translation of `size_t` to `u64` should have been to `usize` and I didn't realise it wasn't, partly due to not completely applying the reductions so far.

fixes #36
see #25

You should be able to add commits on the PR branch.

### CODE REVIEW

**Comment by crlf0710** [comment]
Hi, thanks for the contribution. However i feel the the proposed structure is a little over sparsed. In #56 I renamed the xetex_macos.rs to aatfont.rs. Could you rebase your work(CF/CT/etc) on top of that?
Also for the behavior changes of `gFreeTypeLibrary`, I think we should better leave it to another dedicated PR. You can use the `-sys` crate first if that's easier for dealing with c code.

**Comment by cormacrelf** [comment]
I haven't extracted out the `gFreeTypeLibrary` changes as I have to go do other things, but that's a quick rebase. (For the record / for any subsequent PR extraction, this use of freetype is literally just to get postscript names of font variants, and the original code dropped all the fonts it made so this does too. The only shared thing was the few lines of library initialization which is AFAIK pretty cheap.)

File: engine/src/xetex_aatfont.rs [hunk 1]
```diff
@@ -1,2701 +1,437 @@
-#![cfg(target_os="macos")]
```
Comment (crlf0710) [inline engine/src/xetex_aatfont.rs:1]: Please keep this `cfg` line

**Comment by crlf0710** [comment]
Mostly looks great modulo one comment above.

**Comment by crlf0710** [comment]
There's one compilation error, do you mind me fixing it directly? Or do you want to fix that yourself?

**Comment by cormacrelf** [comment]
I don't mind but I already did it :)

**Pull request merged**